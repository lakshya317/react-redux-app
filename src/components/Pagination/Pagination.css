.pagination-container{
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
}

.pagination-main{
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
}

.page-arrow{
    height: 13px;
    margin: 0px 5px;
}

.page-number{
    font-size: 15px;
    font-weight: 600;
    margin: 0px 5px;
    padding: 10px;
}

.page-arrow:hover, .page-number:hover{
    cursor: pointer;
}

.current-page{
    color: var(--shell-red);
    /* border: 1px solid var(--shell-red); */
    /* border-radius: 10px; */
}

.pagination-summary{
    display: flex;
    align-self: flex-end;
    font-size: 13px;
    font-weight: 600;
    margin: 0px 5px;
    padding: 10px;
}