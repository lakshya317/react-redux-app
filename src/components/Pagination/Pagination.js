import React from 'react';
import "./Pagination.css";
import RightArrow from "../../assets/icons/right-arrow.png";
import LeftArrow from "../../assets/icons/left-arrow.png";
import RightDoubleArrow from "../../assets/icons/right-double-arrow.png";
import LeftDoubleArrow from "../../assets/icons/left-double-arrow.png";

const Pagination = (props) => {
    const {current, perPage, handlePageChange, totalPages, total} = props;

    const pageChange = (newPage) => {
        if(!(newPage < 1 || newPage > totalPages)){
            handlePageChange(newPage);
        }
    }
    return (
        <div className='pagination-container'>
            <div className="pagination-main">
                <img className="page-arrow" src={LeftDoubleArrow} alt={"First"} onClick={() => pageChange(1)} />
                <img className="page-arrow" src={LeftArrow} alt={"Previous"} onClick={() => pageChange(current - 1)} />
                    {
                    Array.from(Array(totalPages).keys()).map((index) => {
                        let page = index + 1;
                        return (
                            <div
                                className={`page-number${page === current ? " current-page" : ""}`}
                                onClick={() => pageChange(page)}
                                key={index}
                            >
                                {page}
    
                            </div>
                        );
    
                    })
                    }
                <img className="page-arrow" src={RightArrow} alt={"Next"} onClick={() => pageChange(current + 1)} />
                <img className="page-arrow" src={RightDoubleArrow} alt={"Last"} onClick={() => pageChange(props.totalPages)} />
            </div>
            <div className="pagination-summary">
                {`Showing ${1+ (perPage * (current-1))} - ${perPage*current} of ${total}`}
            </div>
        </div>
    );
}

export default Pagination;
