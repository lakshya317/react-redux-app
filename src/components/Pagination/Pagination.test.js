import { render } from '@testing-library/react';
import Pagination from './Pagination';
import { MemoryRouter } from 'react-router-dom';

describe('Pagination Tests', () => {

    test('Pagination Snapshot Test', () => {
        const pagination = render(
            <MemoryRouter>
                <Pagination current={2} perPage={6} total={17} totalPages={3} handlePageChange={(newPage)=>{console.log(newPage)}}/>
            </MemoryRouter>
        );

        expect(pagination).toMatchSnapshot()
    });

    test('Pagination Page Numbers and icon verification', () => {
        const pagination = render(
            <MemoryRouter>
                <Pagination current={2} perPage={6} total={17} totalPages={3} handlePageChange={(newPage)=>{console.log(newPage)}}/>
            </MemoryRouter>
        );

        pagination.getByText("1");
        pagination.getByText("2");
        pagination.getByText("3");
        pagination.getByAltText("First");
        pagination.getByAltText("Previous");
        pagination.getByAltText("Next");
        pagination.getByAltText("Last");
    })

    test('Pagination Text verification', () => {
        const pagination = render(
            <MemoryRouter>
                <Pagination current={2} perPage={6} total={17} totalPages={3} handlePageChange={(newPage)=>{console.log(newPage)}}/>
            </MemoryRouter>
        );

        pagination.getByText("Showing 7 - 12 of 17", {exact: false});
    })
})