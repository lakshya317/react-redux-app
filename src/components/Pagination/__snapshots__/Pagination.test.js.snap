// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Tests Pagination Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="pagination-container"
      >
        <div
          class="pagination-main"
        >
          <img
            alt="First"
            class="page-arrow"
            src="left-double-arrow.png"
          />
          <img
            alt="Previous"
            class="page-arrow"
            src="left-arrow.png"
          />
          <div
            class="page-number"
          >
            1
          </div>
          <div
            class="page-number current-page"
          >
            2
          </div>
          <div
            class="page-number"
          >
            3
          </div>
          <img
            alt="Next"
            class="page-arrow"
            src="right-arrow.png"
          />
          <img
            alt="Last"
            class="page-arrow"
            src="right-double-arrow.png"
          />
        </div>
        <div
          class="pagination-summary"
        >
          Showing 7 - 12 of 17
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="pagination-container"
    >
      <div
        class="pagination-main"
      >
        <img
          alt="First"
          class="page-arrow"
          src="left-double-arrow.png"
        />
        <img
          alt="Previous"
          class="page-arrow"
          src="left-arrow.png"
        />
        <div
          class="page-number"
        >
          1
        </div>
        <div
          class="page-number current-page"
        >
          2
        </div>
        <div
          class="page-number"
        >
          3
        </div>
        <img
          alt="Next"
          class="page-arrow"
          src="right-arrow.png"
        />
        <img
          alt="Last"
          class="page-arrow"
          src="right-double-arrow.png"
        />
      </div>
      <div
        class="pagination-summary"
      >
        Showing 7 - 12 of 17
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
